cmd/upgrade_test.go

```diff
@@ -4,104 +4,139 @@
+			expectedErr: errors.New("--location must be specified"),
+		},
+		{
+			uc: &upgradeCmd{
+				resourceGroupName:   "test",
+				deploymentDirectory: "_output/test",
+				upgradeVersion:      "",
+				location:            "southcentralus",
+				timeoutInMinutes:    60,
 			},
-			{
-				uc: &upgradeCmd{
-					resourceGroupName:   "test",
-					deploymentDirectory: "",
-					upgradeVersion:      "1.9.0",
-					location:            "southcentralus",
-					timeoutInMinutes:    60,
-				},
-				expectedErr: errors.New("--deployment-dir must be specified"),
+			expectedErr: errors.New("--upgrade-version must be specified"),
+		},
+		{
+			uc: &upgradeCmd{
+				resourceGroupName:   "test",
+				deploymentDirectory: "",
+				upgradeVersion:      "1.9.0",
+				location:            "southcentralus",
+				timeoutInMinutes:    60,
 			},
-			{
-				uc: &upgradeCmd{
-					resourceGroupName:   "test",
-					deploymentDirectory: "",
-					upgradeVersion:      "1.9.0",
-					location:            "southcentralus",
-					timeoutInMinutes:    60,
-				},
-				expectedErr: errors.New("--deployment-dir must be specified"),
+			expectedErr: errors.New("--deployment-dir must be specified"),
+		},
+		{
+			uc: &upgradeCmd{
+				resourceGroupName:   "test",
+				deploymentDirectory: "",
+				upgradeVersion:      "1.9.0",
+				location:            "southcentralus",
+				timeoutInMinutes:    60,
 			},
-			{
-				uc: &upgradeCmd{
-					resourceGroupName:   "test",
-					deploymentDirectory: "_output/mydir",
-					upgradeVersion:      "1.9.0",
-					location:            "southcentralus",
-				},
-				expectedErr: nil,
+			expectedErr: errors.New("--deployment-dir must be specified"),
+		},
+		{
+			uc: &upgradeCmd{
+				resourceGroupName:   "test",
+				deploymentDirectory: "_output/mydir",
+				upgradeVersion:      "1.9.0",
+				location:            "southcentralus",
 			},
-		}
+			expectedErr: nil,
+		},
+	}
 
-		for _, c := range cases {
-			err := c.uc.validate(r)
-			if c.expectedErr != nil && err != nil {
-				Expect(err.Error()).To(Equal(c.expectedErr.Error()))
-			} else {
-				Expect(err).To(BeNil())
-				Expect(c.expectedErr).To(BeNil())
-			}
+	for _, c := range cases {
+		err := c.uc.validate(r)
+		if c.expectedErr != nil && err != nil {
+			g.Expect(err.Error()).To(Equal(c.expectedErr.Error()))
+		} else {
+			g.Expect(err).To(BeNil())
+			g.Expect(c.expectedErr).To(BeNil())
 		}
+	}
+}
+
+func TestCreateUpgradeCommandSuccesfully(t *testing.T) {
+	g := NewGomegaWithT(t)
+	output := newUpgradeCmd()
+
+	g.Expect(output.Use).Should(Equal(upgradeName))
+	g.Expect(output.Short).Should(Equal(upgradeShortDescription))
+	g.Expect(output.Long).Should(Equal(upgradeLongDescription))
+	g.Expect(output.Flags().Lookup("location")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("resource-group")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("deployment-dir")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("upgrade-version")).NotTo(BeNil())
+}
+
```

Comment:
it is clearer now that we will authorize same version upgrade only when the --force parameter is provided